File: 02_two_pointers/02_two_sum_ii/solution.py
"""
Two Pointers -> Two Sum II - Input Array is Sorted
Neetcode 11 / 150
Leetcode 167

Python

Comments:
So fun, so simple!

Time: O(n) the two pointers traverse whole list at most once
Space: O(1) we don't need to make anything new
"""


class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        l = 0
        r = len(numbers) - 1
        while l < r:
            total = numbers[l] + numbers[r]
            if total == target:
                return [l + 1, r + 1]
            elif total < target:
                l += 1
            else:
                r -= 1
        return []  # return empty list if no solution found

File: 02_two_pointers/02_two_sum_ii/solution.js
/*
Two Pointers -> Two Sum II - Input Array is Sorted
Neetcode 11 / 150
Leetcode 167

Javascript

Comments:
So fun, so simple!

Time: O(n) the two pointers traverse whole array at most once
Space: O(1) we don't need to make anything new
*/

class Solution {
	/**
	 * @param {number[]} numbers
	 * @param {number} target
	 * @return {number[]}
	 */
	twoSum(numbers, target) {
		let l = 0;
		let r = numbers.length - 1;
		while (l < r) {
			const total = numbers[l] + numbers[r];
			if (total === target) {
				return [l + 1, r + 1];
			} else if (total < target) {
				l++;
			} else {
				r--;
			}
		}
		return []; // return empty array if no solution found
	}
}
